novice-r-practice.Rmd

```diff
@@ -0,0 +1,363 @@
+  dplyr::group_by(quantity) %>%
+  dplyr::count()
+```
+
+*Group by person and quantity measured.*
+
+4. Find the minimum, average, and maximum for each quantity:
+
+```{r min-ave-max}
+cleaned %>%
+  dplyr::group_by(quantity) %>%
+  dplyr::summarize(low = min(reading), mid = mean(reading), high = max(reading))
+```
+
+*Look at the range for each combination of person and quantity.*
+
+5. Rescale salinity measurements that are greater than 1:
+
+```{r rescale-salinity}
+cleaned <- cleaned %>%
+  dplyr::mutate(reading = ifelse(quantity == 'sal' & reading > 1.0, reading/100, reading))
+cleaned
+```
+
+*Do the same calculation use `case_when`.*
+
+6. Read `visited.csv`, drop the NAs, and join with the cleaned-up table of readings:
+
+```{r join-two-tables}
+cleaned <- readr::read_csv(here::here("novice-r-practice", "visited.csv")) %>%
+  dplyr::filter(!is.na(visit_date)) %>%
+  dplyr::inner_join(cleaned, by = c("visit_id" = "visit_id"))
+cleaned
+```
+
+*Join `visited.csv` with `site.csv` to get (date, latitude, longitude) triples for site visits.*
+
+7. Find the dates of the highest radiation reading at each site:
+
+```{r dates-high-rad}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(max_rad = max(reading)) %>%
+  dplyr::filter(reading == max_rad)
+```
+
+Another way to do it:
+
+```{r dates-high-rad-2}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::top_n(1, reading) %>%
+  dplyr::select(site_id, visit_date, reading)
+```
+
+*Explain why this __doesn't__ work.*
+
+```{r error-high-rad, error=TRUE}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::summarize(max_rad = max(reading)) %>%
+  dplyr::ungroup() %>%
+  dplyr::filter(reading == max_rad)
+```
+
+8. Normalize radiation against the highest radiation seen per site:
+
+```{r normalize-rad}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(
+    max_rad = max(reading),
+    frac_rad = reading / max_rad) %>%
+  dplyr::select(visit_id, site_id, visit_date, frac_rad)
+```
+
+*Normalize salinity against mean salinity by site.*
+
+9. Find stepwise change in radiation per site by date:
+
+```{r rad-change}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::arrange(site_id, visit_date)
+```
+
+*Find length of time between visits by site.*
+
+10. Find sites that experience any stepwise increase in radiation between visits:
+
+```{r rad-increases}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::filter(!is.na(delta_rad)) %>%
+  dplyr::summarize(any_increase = any(delta_rad > 0)) %>%
+  dplyr::filter(any_increase)
+```
+
+*Find sites with visits more than one year apart.*
+
+## Creating charts
```

Comment:
Nothing in this plotting section relies on previous data manipulation sections.  I'd love to see visualization first, then exercises above could use plots to verify results.

Reply:
I think we'd need a larger dataset for that - this one is fabricated and only has a handful of entries. (I wanted to keep it small so that learners could hand-check results, but maybe that's a mistake?)

Reply:
I think inside a single lesson small hand-checkable examples are great!  And these are perfect for that.  

I was just thinking for exercises that synthesize over a few lessons, it would be nice to have tasks that pull from earlier sections, and to emphasize visualization as a tool throughout the analysis process, not just at the end.